### PR TITLE
Muffle updates to the cursor position to 1/~100ms

### LINKED ISSUE #5288: TerminalCore/TerminalControl dispatch five billion cursor update coroutines when printing long file
When we're printing a long file, we call `winrt::fire_and_forget TermControl::_TerminalCursorPositionChanged()` an absolutely mad number of times. Each one of them unconditionally dispatches and waits to tell the TSF input control it's ready to dispatch a cursor update.

We can probably do this only once per dispatcher cycle.

### PR DESCRIPTION
**Pull request by DHowett-MSFT**
This stops us from dispatching back-to-back terminal cursor position
updates to the TSF control before it has a chance to get back to us.

Fixes #5288 

This was tested live with the 6MB repro file from "Terminal Not Speed Enough".

### CODE REVIEW

**Comment by miniksa** [comment]
I still wanted it to be named ["tuned mass damper"](https://www.popularmechanics.com/technology/infrastructure/a14564/the-121-story-tower-that-never-sways/) but fine. Let's do this.

**Comment by DHowett-MSFT** [comment]
We still dispatch about 60 of them printing the entire 6MB file, but that's a 100000000000000000% reduction and the UI remains fully responsive.

**Comment by DHowett-MSFT** [comment]
Alright, i switched this from a mutex to an atomic -- it's cheaper and the platform cannot deny us.

I also took some measurements. For a .070-second run of trace logging from that 6MB file:

We tried to notify TSF 19798 times, and we successfully notified it 14 times.






We can probably do better than "notify 14 times in 7 hundredths of a second"

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 1]
```diff
@@ -1847,13 +1847,25 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
...
+
         if (_closing.load())
         {
             return;
         }
 
         auto weakThis{ get_weak() };
 
+        // Even with muffling, we dispatch this tens of times _per hundredth of a second_.
+        // Muffle it harder.
```
Comment (miniksa) [inline src/cascadia/TerminalControl/TermControl.cpp:1865]:
```suggestion
        // Muffle Hard 2: Muffle Harder.

```

Comment (DHowett-MSFT) [reply src/cascadia/TerminalControl/TermControl.cpp:1865]: rofl.

File: src/cascadia/TerminalControl/TermControl.cpp [hunk 2]
```diff
@@ -1847,13 +1847,25 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
...
+        static constexpr auto CursorUpdateQuiesceTime{ std::chrono::milliseconds(100) };
+        co_await winrt::resume_after(CursorUpdateQuiesceTime);
```
Comment (miniksa) [inline src/cascadia/TerminalControl/TermControl.cpp:1868]: Sorry, can you explain in plain English why this works?

It takes me effort to discern that there's one of them stuck in here with the state sitting at TRUE above that will reject all the other updates even if this one is about to be serviced just to slow it down.

Looking for something like 

// If we're the lucky one who gets through, we'll still wait 100ms to clog the atomic above so we don't service the cursor update too fast. If we get through and finish processing the update quickly but similar requests are still beating down the door above in the atomic, we may still update the cursor way more than is visible to anyone's eye, which is a waste of effort.

**Comment by DHowett-MSFT** [comment]
After damping, i got it down to 502 cursor updates for the entire file print.

File: src/cascadia/TerminalControl/TermControl.cpp (hunk 2 above)
Comment (DHowett-MSFT) [reply src/cascadia/TerminalControl/TermControl.cpp:1868]: I'll just wholesale copy this comment, it's great. thanks

**Comment by zadjii-msft** [comment]
Hey I'm just seeing this now in the morning, but @greg904 had a great little throttle helper over in [`TimeThrottle.h`](https://github.com/microsoft/terminal/pull/4608/files?file-filters%5B%5D=.h&file-filters%5B%5D=.vcxproj#diff-0254c79502dc99a9f37a43d150b43584) in #4608 that might be able to be adapted for this to dampen even further
